ogr/ogrsf_frmts/miramon/mm_wrlayr.c

```diff
@@ -0,0 +1,7038 @@
+int MMResizePolHeaderPointer(struct MM_PH **pPolHeader, GUInt64 *nMax,
+                             GUInt64 nNum, GUInt64 nIncr, GUInt64 nProposedMax);
+void MMUpdateBoundingBoxXY(struct MMBoundingBox *dfBB,
+                           struct MM_POINT_2D *pCoord);
+void MMUpdateBoundingBox(struct MMBoundingBox *dfBBToBeAct,
+                         struct MMBoundingBox *dfBBWithData);
+int MMCheckVersionFor3DOffset(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                              MM_FILE_OFFSET nOffset,
+                              MM_INTERNAL_FID nElemCount);
+int MMCheckVersionOffset(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                         MM_FILE_OFFSET OffsetToCheck);
+int MMCheckVersionForFID(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                         MM_INTERNAL_FID FID);
+
+// Extended DBF functions
+int MMCreateMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer);
+int MMAddDBFRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                         struct MiraMonFeature *hMMFeature);
+int MMAddPointRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                           struct MiraMonFeature *hMMFeature,
+                           MM_INTERNAL_FID nElemCount);
+int MMAddArcRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                         struct MiraMonFeature *hMMFeature,
+                         MM_INTERNAL_FID nElemCount, struct MM_AH *pArcHeader);
+int MMAddNodeRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                          MM_INTERNAL_FID nElemCount,
+                          struct MM_NH *pNodeHeader);
+int MMAddPolygonRecordToMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer,
+                             struct MiraMonFeature *hMMFeature,
+                             MM_INTERNAL_FID nElemCount,
+                             MM_N_VERTICES_TYPE nVerticesCount,
+                             struct MM_PH *pPolHeader);
+int MMCloseMMBD_XP(struct MiraMonVectLayerInfo *hMiraMonLayer);
+void MMDestroyMMDB(struct MiraMonVectLayerInfo *hMiraMonLayer);
+
+/* -------------------------------------------------------------------- */
+/*      Managing errors and warnings                                    */
+/* -------------------------------------------------------------------- */
+
+void MMCPLError(int level, int code, const char *message)
+{
+#ifdef GDAL_COMPILATION
+    CPLError(level, code, "%s", message);
+#else
+    sprintf(local_message, "level(%d), code(%d)", level, code);
+    InfoMsg(local_message);
+    ErrorMsg(message);
+#endif
+}
+
+void MMCPLWarning(int level, int code, const char *message)
+{
+#ifdef GDAL_COMPILATION
+    CPLError(level, code, "%s", message);
+#else
+    sprintf(local_message, "level(%d), code(%d)", level, code);
+    InfoMsg(local_message);
+    InfoMsg(message);
+#endif
+}
+
+void MMCPLDebug(const char *c, const char *message)
+{
+#ifdef GDAL_COMPILATION
+    CPLDebug(c, "%s", message);
```

Comment:
As there's quite a large number of debug messages in the code, that could alter the performance for release builds. So I'd suggest to map to CPLDebugOnly() which will evaluate to CPLDebug() for debug builds, and to nothing for release builds:

```suggestion
    CPLDebugOnly(c, "%s", message);
```

Reply:
Oh, I didn't know it. Thanks. Done